Client/src/Components/TabPanels/Status/GeneralSettingsPanel.jsx

```diff
@@ -0,0 +1,281 @@
+		const { error } = publicPageGeneralSettingsValidation.validate(
+			{ [id]: value },
+			{
+				abortEarly: false,
+			}
+		);
+
+		setErrors((prev) => {
+			return buildErrors(prev, id, error);
+		});
+	};
+
+	const validateField = (toValidate, schema, name = "logo") => {
+		const { error } = schema.validate(toValidate, { abortEarly: false });
+		setErrors((prev) => {
+			let prevErrors = { ...prev };
+			if (error) prevErrors[name] = error?.details[0].message;
+			else delete prevErrors[name];
+			return prevErrors;
+		});
+		if (error) return true;
+	};
+
+	const handleLogo = (event) => {
+		const pic = event.target?.files?.[0];
+		let error = validateField({ type: pic?.type, size: pic?.size }, logoImageValidation);
+		if (error) return;
+
+		setProgress((prev) => ({ ...prev, isLoading: true }));
+		setLogo({
+			src: URL.createObjectURL(pic),
+			name: pic.name,
+			type: pic.type,
+			size: pic.size,
+		});
+		intervalRef.current = setInterval(() => {
+			const buffer = 12;
+			setProgress((prev) => {
+				if (prev.value + buffer >= 100) {
+					clearInterval(intervalRef.current);
+					return { value: 100, isLoading: false };
+				}
+				return { ...prev, value: prev.value + buffer };
+			});
+		}, 120);
+	};
+
+	return (
+		<TabPanel
+			value="General Settings"
+			sx={{
+				"& h1, & p, & input": {
+					color: theme.palette.text.tertiary,
+				},
+			}}
+		>
+			<Stack
+				component="form"
+				className="status-general-settings-form"
+				noValidate
+				spellCheck="false"
+				gap={theme.spacing(12)}
+				mt={theme.spacing(6)}
+			>
+				<ConfigBox>
+					<Box>
+						<Stack gap={theme.spacing(6)}>
+							<Typography component="h2">Access</Typography>
+							<Typography component="p">
+								If your status page is ready, you can mark it as published.
+							</Typography>
+						</Stack>
+					</Box>
+					<Stack gap={theme.spacing(6)}>
+						<Checkbox
+							id="published-to-public"
+							label={`Published and visible to the public`}
+							isChecked={localData.publish}
+							value={"localData.publish"}
+							onChange={handleChange}
+							onBlur={handleBlur}
+						/>
+					</Stack>
+				</ConfigBox>
+
+				<ConfigBox>
+					<Box>
+						<Stack gap={theme.spacing(6)}>
+							<Typography component="h2">Basic Information</Typography>
+							<Typography component="p">
+								Define company name and the subdomain that your status page points to.
+							</Typography>
+						</Stack>
+					</Box>
+					<Stack gap={theme.spacing(18)}>
+						<TextInput
+							id="companyName"
+							type="text"
+							label="Company name"
+							value={localData.companyName}
+							onChange={handleChange}
+							onBlur={handleBlur}
+							helperText={errors["companyName"]}
+							error={errors["companyName"] ? true : false}
+						/>
+						<TextInput
+							id="url"
+							type="url"
+							label="SubURL"
+							value={localData.url}
+							startAdornment={<HttpAdornment https={true} />}
```

Comment:
This won't necessarily be HTTPS, we should provide an http/https selector as in the Create Monitors page

Reply:
yes, fixed in next branch: Feat/fe/Status page 3